warehouse/index-stats/src/main/java/datawave/mapreduce/shardStats/HyperLogFieldSummary.java

```diff
@@ -0,0 +1,70 @@
+package datawave.mapreduce.shardStats;
+
+import com.clearspring.analytics.stream.cardinality.CardinalityMergeException;
+import com.clearspring.analytics.stream.cardinality.HyperLogLogPlus;
+import org.apache.accumulo.core.data.Value;
+
+import java.io.IOException;
+import java.util.Objects;
+
+/**
+ * POJO fo summary data for each field name/datatype pair processed by the reducer.
+ */
+class HyperLogFieldSummary implements FieldSummary {
+    
+    private long count;
+    private final HyperLogLogPlus logPlus;
+    
+    HyperLogFieldSummary(HyperLogLogPlus hllp) {
+        this.logPlus = hllp;
+    }
+    
+    @Override
+    public long getCount() {
+        return this.count;
+    }
+    
+    @Override
+    public void add(Value value) throws IOException {
+        StatsHyperLogSummary stats = new StatsHyperLogSummary(value);
+        this.count += stats.getCount();
+        HyperLogLogPlus hllpAdd = stats.getHyperLogPlus();
+        try {
+            this.logPlus.addAll(hllpAdd);
+        } catch (CardinalityMergeException e) {
+            // addAll throws an out of scope exception
+            throw new IOException();
```

Comment:
Can we propagate the CardinalityMergeException to preserve the full stacktrace?

Reply:
Yes, I meant to add that as the cause.